### ISSUE TITLE
Bug In readTokenDecryptionKey

### ISSUE THREAD

**Issue by jainhitesh9998** [quarkusio/quarkus#33021]
### Describe the bug
Module - quarkus-oidc
https://github.com/quarkusio/quarkus/blob/9380d094596ddf7ba6f007c9e58c1f1781722881/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java#L327

this condition resolved to false even when i pass a valid jwks private key 
`keys.get(0).getAlgorithm() == KeyEncryptionAlgorithm.RSA_OAEP.getAlgorithm()`, resulting in application startup failure on configuration of this  QUARKUS_OIDC_TOKEN_DECRYPTION_KEY_LOCATION parameter

I've changed it to in my local builds and the resolution happens properly and application starts up 
`properlykeys.get(0).getAlgorithm().equals(KeyEncryptionAlgorithm.RSA_OAEP.getAlgorithm())`

### Expected behavior

The resolution should be true and if jwks is passed and the private key should be set

### Actual behavior

since this check fails, the flow moves to decoding the keyContent as pem file and it gives error in parsing base64 string
```
if (key == null) {
                    key = KeyUtils.decodeDecryptionPrivateKey(keyContent);
                }
```

### How to Reproduce?

I'm trying to configure and use jwe with quarkus-oidc

configure QUARKUS_OIDC_TOKEN_DECRYPTION_KEY_LOCATION with a location of a private jwks key along with other oidc properties.

I can provide the complete setup i'm trying to validate if necessary.

This looks like a trivial bug, and i can send a PR for this.

### Output of `uname -a` or `ver`

 21.6.0 Darwin Kernel Version 21.6.0: Mon Dec 19 20:43:09 PST 2022; root:xnu-8020.240.18~2/RELEASE_ARM64_T6000 arm64classes/ 

### Output of `java -version`

openjdk 17.0.4.1 2022-08-12 LTS

### Quarkus version or git rev

2.16

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.5

**Comment by sberyozkin** [comment]
@jainhitesh9998 Thanks, the test uses the pem file as far as I recall, would you to open a simply PR to fix it ?

**Comment by jainhitesh9998** [comment]
The private jwk Key  is configured in this tests:
 `quarkus.oidc.code-flow-encrypted-id-token-jwk.token.decryption-key-location=privateKey.jwk 
`

https://github.com/quarkusio/quarkus/blob/main/test-framework/oidc-server/src/main/resources/privateKey.jwk this file doesn't contain 'alg' and 'use' field

 fixing the properties and adding another PR to validate the changes

**Comment by jainhitesh9998** [comment]
As of now i only see one enc type supported "RSA-OAEP" 
RSA-OAEP-256 and a few other variants were missing, jwk with those specific enc types would also result in error due going to pem workflow, wouldn't it be better to throw an exception for unsupported type, there

**Comment by sberyozkin** [comment]
Closed by #33023
